python_modules/jupedsim/jupedsim/distributions.py

```diff
@@ -313,11 +341,20 @@ def distribute_by_percentage(*, polygon, percent, distance_to_agents, distance_t
         :param seed: define a seed for random generation, Default value is None which corresponds to a random value
         :param max_iterations: no more than max_iterations must find a point inside the polygon, Default is 10_000
         :return: list of created points"""
-    samples = distribute_til_full(polygon=polygon, distance_to_agents=distance_to_agents,
-                                  distance_to_polygon=distance_to_polygon, seed=seed, max_iterations=max_iterations)
+    samples = distribute_till_full(polygon=polygon, distance_to_agents=distance_to_agents,
+                                  distance_to_polygon=distance_to_polygon, seed=seed,
+                                  max_iterations=max_iterations, k=k)
     sample_amount = len(samples)
-    needed_amount = round(sample_amount * (percent/100))
-    return samples[:needed_amount]
+    needed_amount = round(sample_amount * (percent / 100))
+    np.random.seed(seed)
+    selected_samples = []
+    for i in range(needed_amount):
+        temp_sample = samples[np.random.randint(0, len(samples))]
+        selected_samples.append(temp_sample)
+        samples.remove(temp_sample)
```

Comment:
Why can't you just do it the way before: `samples[:needed_amount]`?

Reply:
the next points always depend on the previously placed points.
So taking only the first points would create a cluster around the first placed point.

Reply:
It might be better to shuffle the list and then just return `samples[:needed_amount]`

Reply:
done